packages/fe/components/cid-table.vue

```diff
@@ -53,30 +110,80 @@ export default {
   name: 'CIDTable',
 
   components: {
-    CIDCard
+    CIDCard,
+    PaginationControls,
+    Limit,
+    Searchbar,
+    FieldContainer,
+    Spinner
   },
 
   data () {
     return {
       mobileTable: false,
-      resize: false
+      resize: false,
+      initialFetchFinished: false,
+      checkboxes: [
+        { label: 'Show only complete data', value: false },
+        { label: 'Show only available CIDs', value: false }
+      ]
     }
   },
 
   computed: {
     ...mapGetters({
-      cidList: 'dataset/cidList'
-    })
+      cidList: 'cid/cidList',
+      metadata: 'cid/metadata',
+      cidsLoading: 'cid/loading',
+      limitOptions: 'datasets/limitOptions',
+      filters: 'datasets/filters'
+    }),
+    totalPages () {
+      return this.metadata.totalPages
+    },
+    page () {
+      return this.metadata.page
+    },
+    resultsCount () {
+      return this.metadata.count
+    }
+  },
+
+  watch: {
+    '$route' () {
+      this.$nextTick(() => {
+        this.getCidList({ route: this.$route })
+      })
+    },
+    cidList () {
+      this.stopLoading()
+    }
   },
 
-  mounted () {
+  async mounted () {
     handleTableResize(this)
     this.resize = () => { handleTableResize(this) }
     window.addEventListener('resize', this.resize)
+    await this.getCidList({ route: this.$route })
```

Comment:
All data fetching should (in 99% of cases) happen on the page level. Let's move this to `_id.vue`'s mounted hook